src/main/java/com/microsoft/alm/authentication/KeychainSecurityCliStore.java

```diff
@@ -304,6 +307,22 @@ else if ("blob".equals(type.toString()))
         // TODO: else if ("sint32".equals(type))
     }
 
+    public boolean isKeychainAvailable()
+    {
+        final String targetName = "test:isKeychainAvailable";
+        final Token token = new Token("this is a test token", TokenType.Test);
+        try
+        {
+            writeToken(targetName, token);
+        }
+        catch (final SecurityException e)
+        {
+            return false;
+        }
+        delete(targetName);
+        return true;
+    }
```

Comment:
Looking at other security commands, it appears show-keychain-info has no side-effects, yet exits with code 36 when "User interaction is not allowed."; I'll try using that.

Sounds great!  Fingers crossed... thanks.